lib/msal-common/test/authority/AuthorityFactory.spec.ts

```diff
@@ -24,49 +25,55 @@ describe("AuthorityFactory.ts Class Unit Tests", () => {
-                networkInterface
+                networkInterface,
+                ProtocolMode.AAD
             )
         ).to.throw(ClientConfigurationErrorMessage.authorityUriInsecure.desc);
-        expect(() =>
-            AuthorityFactory.createInstance(
-                "https://login.microsoftonline.com/",
-                networkInterface
-            )
-        ).to.throw(ClientConfigurationErrorMessage.urlParseError.desc);
         expect(() =>
             AuthorityFactory.createInstance(
                 "This is not a URI",
-                networkInterface
+                networkInterface,
+                ProtocolMode.AAD
             )
         ).to.throw(ClientConfigurationErrorMessage.urlParseError.desc);
         expect(() =>
-            AuthorityFactory.createInstance("", networkInterface)
+            AuthorityFactory.createInstance("", networkInterface, ProtocolMode.AAD)
         ).to.throw(ClientConfigurationErrorMessage.urlEmptyError.desc);
     });
 
     it("createInstance returns Default instance if AAD Authority", () => {
-        const authorityInstance = AuthorityFactory.createInstance(Constants.DEFAULT_AUTHORITY, networkInterface);
+        const authorityInstance = AuthorityFactory.createInstance(Constants.DEFAULT_AUTHORITY, networkInterface, ProtocolMode.AAD);
         expect(authorityInstance.authorityType).to.be.eq(AuthorityType.Default);
         expect(authorityInstance instanceof Authority);
     });
 
     it("createInstance returns Default instance if B2C Authority", () => {
-        const authorityInstance = AuthorityFactory.createInstance(TEST_CONFIG.b2cValidAuthority, networkInterface);
+        const authorityInstance = AuthorityFactory.createInstance(TEST_CONFIG.b2cValidAuthority, networkInterface, ProtocolMode.AAD);
         expect(authorityInstance.authorityType).to.be.eq(AuthorityType.Default);
         expect(authorityInstance instanceof Authority);
     });
 
     it("createInstance return ADFS instance if /adfs in path", () => {
-        const authorityInstance = AuthorityFactory.createInstance(TEST_CONFIG.ADFS_VALID_AUTHORITY, networkInterface);
-        expect(authorityInstance.authorityType).to.be.eq(AuthorityType.Adfs);
-        expect(authorityInstance instanceof Authority);
+        const authorityInstanceAAD = AuthorityFactory.createInstance(TEST_CONFIG.ADFS_VALID_AUTHORITY, networkInterface, ProtocolMode.AAD);
+        expect(authorityInstanceAAD.authorityType).to.be.eq(AuthorityType.Adfs);
+        expect(authorityInstanceAAD instanceof Authority);
+
+        const authorityInstanceOIDC = AuthorityFactory.createInstance(TEST_CONFIG.ADFS_VALID_AUTHORITY, networkInterface, ProtocolMode.OIDC);
+        expect(authorityInstanceOIDC.authorityType).to.be.eq(AuthorityType.Adfs);
+        expect(authorityInstanceOIDC instanceof Authority);
     });
+
+    it("createInstance returns (non v2) OIDC endpoint with ProtocolMode: OIDC", () => {
```

Comment:
What is "non-v2 OIDC" as opposed to just "OIDC"? Does this mean "Not azure AD v2"?

Reply:
I think ProtocolMode.AAD => non v2.0 endpoints (i.e. endpoints that don't have v2.0 before the `/.well-known/openid-configuration` for the discovery endpt)

Reply:
It has to do with how the well-known endpoint is constructed by appending `v2.0/` in the path, as shown in my reply to your other comment. This could also probably be clearer.